paimon-core/src/main/java/org/apache/paimon/mergetree/LookupLevels.java

```diff
@@ -102,26 +103,59 @@ public void notifyDropFile(String file) {
     }
 
     @Nullable
-    public KeyValue lookup(InternalRow key, int startLevel) throws IOException {
-        if (startLevel == 0) {
-            throw new IllegalArgumentException("Start level can not be zero.");
-        }
-
+    public KeyValue lookup(LookupContext context) throws IOException {
         KeyValue kv = null;
-        for (int i = startLevel; i < levels.numberOfLevels(); i++) {
-            SortedRun level = levels.runOfLevel(i);
-            kv = lookup(key, level);
-            if (kv != null) {
-                break;
-            }
-        }
 
-        return kv;
+        switch (context.order) {
+            case High2Lower:
```

Comment:
Can you explain why here using `High2Lower` and `Low2Higher`? My understanding is that we can just check key contains.

Reply:
I think you are right, this solution is not necessary, will refactor this part.